src/librawspeed/decoders/Cr2Decoder.cpp

```diff
@@ -131,11 +130,10 @@ RawImage Cr2Decoder::decodeNewFormat() {
   TiffEntry* offsets = raw->getEntry(STRIPOFFSETS);
   TiffEntry* counts = raw->getEntry(STRIPBYTECOUNTS);
 
-  LJpegPlain l(*mFile, offsets->getInt(), counts->getInt(), mRaw);
-  l.addSlices(s_width);
+  Cr2Decompressor d(*mFile, offsets->getInt(), counts->getInt(), mRaw);
```

Comment:
All these decompressors have basically the same constructor, which makes sense.
There should be `AbstractDecompressor`, similar to the `AbstractTiffDecoder`